apps/gulpfile.js

```diff
@@ -0,0 +1,200 @@
+
+gulp.task('compress', ['browserify'], function () {
+  var uglify = require('gulp-uglify');
+  return gulp.src([
+    './build/package/js/*.js',
+    '!build/package/js/**/vendor.js'
+  ])
+    .pipe(uglify({compress:false}))
+    .pipe(gulp.dest('./build/package/js'));
+});
+
+// MessageFormat .json to .js message processing. Currently specific to apps-modules.
+var MESSAGES_PATH = 'i18n/**/*.json';
+gulp.task('messages', function () {
+  var messageFormat = require('./lib/gulp/transform-messageformat');
+  return gulp.src(MESSAGES_PATH)
+    .pipe(rename(function (filepath) {
+      var app = filepath.dirname;
+      var locale = filepath.basename;
+      filepath.extname = '.js';
+      filepath.dirname = locale;
+      filepath.basename = app + '_locale';
+    }))
+    .pipe(newer(outputDir))
+    .pipe(messageFormat())
+    .pipe(gulp.dest(outputDir));
+});
+
+var ext = 'compressed';
+var vendorFiles = [
+  'lib/blockly/blockly_' + ext + '.js',
+  'lib/blockly/blocks_' + ext + '.js',
+  'lib/blockly/javascript_' + ext + '.js',
+];
+
+var x = {
+  cwd: 'lib/blockly',
+  src: ['??_??.js'],
+  dest: 'build/package/js',
+  // e.g., ar_sa.js -> ar_sa/blockly_locale.js
+  rename: function (dest, src) {
+    var outputPath = src.replace(/(.{2}_.{2})\.js/g, '$1/blockly_locale.js');
+    return path.join(dest, outputPath);
+  }
+};
+
+gulp.task('blockly_locale', function() {
+  var path = require('path');
+  return gulp.src('lib/blockly/??_??.js')
+    .pipe(rename(function (filepath) {
+      // e.g., ./ar_sa.js -> ./ar_sa/blockly_locale.js
+      filepath.dirname = path.join(filepath.dirname, filepath.basename);
+      filepath.basename = 'blockly_locale';
+    }))
+    .pipe(gulp.dest('./build/package/js'));
+});
+
+gulp.task('vendor', ['blockly_locale'], function () {
+  var concat = require('gulp-concat');
+  return gulp.src(vendorFiles)
+    .pipe(newer('./build/package/js/blockly.js'))
+    .pipe(concat('blockly.js'))
+    .pipe(gulp.dest('./build/package/js'));
+});
+
+// Synchronize static files to a public folder
+gulp.task('media', function () {
+  var media = config.media;
+  var mediaStreams = Object.keys(media).map(function(src) {
+    var dest = media[src];
+    return gulp.src(src)
+      .pipe(newer(dest))
+      .pipe(gulp.dest(dest));
+  });
+  return es.merge(mediaStreams);
+});
+
+// Process sass stylesheets to .css
+gulp.task('sass', function () {
+  var sass = require('gulp-sass');
+  var styles = config.stylesheets;
+  var sassStreams = Object.keys(styles).map(function(src) {
+    var dest = styles[src];
+    return gulp.src(src)
+      .pipe(sass())
+      .pipe(newer(dest))
+      .pipe(gulp.dest(dest));
+  });
+  return es.merge(sassStreams);
+});
+
+// Create lodash custom build
+gulp.task('lodash', function (cb) {
+  var exec = require('child_process').exec;
+  return exec('`npm bin`/lodash include="debounce,reject,map,value,range,without,sample,create,flatten,isEmpty,wrap,size,bind" --output src/lodash.js', cb);
+});
```

Comment:
At this point we could arguably get rid of our custom lodash and just pick up the library since we're using a fair amount of it. Not too sure it matters much either way, but removing would get rid of one additional minor complication in our process.